monkey/monkey_island/cc/repository/i_agent_repository.py

```diff
@@ -40,6 +40,16 @@ def get_running_agents(self) -> Sequence[Agent]:
         :raises RetrievalError: If an error occurred while attempting to retrieve the `Agents`
         """
 
+    @abstractmethod
+    def get_progenitor(self, descendant: Agent) -> Agent:
```

Comment:
We should probably just name the parameter 'agent'. Descendant means that it's not the progenitor, but in fact this method might get called where `descendant` IS the progenitor. It's pretty clear from the signature `get_progenitor(self, agent)` that you're getting a progenitor of an agent. IF not, then let's change the name to `get_agent_progenitor`